File: 0021-merge-two-sorted-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(-1);
        ListNode temp = dummy;
        while(list1 != null && list2 != null) {
            if(list1.val <= list2.val) {
                ListNode node1 = new ListNode(list1.val);
                dummy.next = node1;
                list1 = list1.next;
            } else {
                ListNode node2 = new ListNode(list2.val);
                dummy.next = node2;
                list2 = list2.next;
            }
            dummy = dummy.next;
        }

        while(list1 != null) {
            ListNode node1 = new ListNode(list1.val);
            dummy.next = node1;
            list1 = list1.next;
            dummy = dummy.next;
        }

        while(list2 != null) {
            ListNode node2 = new ListNode(list2.val);
            dummy.next = node2;
            list2 = list2.next;
            dummy = dummy.next;
        }
        return temp.next;
    }
}

File: 0021-merge-two-sorted-lists/solution.swift
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init() { self.val = 0; self.next = nil; }
 *     public init(_ val: Int) { self.val = val; self.next = nil; }
 *     public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
 * }
 */
class Solution {
    func mergeTwoLists(_ list1: ListNode?, _ list2: ListNode?) -> ListNode? {
        var dummy = ListNode(-1)
        var temp = dummy

        var l1 = list1
        var l2 = list2

        while l1 != nil && l2 != nil {
            if l1!.val <= l2!.val {
                temp.next = l1
                temp = temp.next!
                l1 = l1?.next
            } else {
                temp.next = l2
                temp = temp.next!
                l2 = l2?.next
            }
        }

        while l1 != nil {
            temp.next = l1
            temp = temp.next!
            l1 = l1?.next
        }

        while l2 != nil {
            temp.next = l2
            temp = temp.next!
            l2 = l2?.next
        }
        return dummy.next
    }
}
